File: Homework/JavaScript-Fundamentals/08. Functions/appearance-count.js
function solve(args) {
    let len = +(args.shift());
    let target = +(args.pop());
    args = args[0].split(' ');

    console.log(countAppearances(args, target))

    function countAppearances(source, target) {
        source = source | [];

        let len = args.length;
        let count = 0;
        for (let i = 0; i < len; i++) {
            if (args[i] == target) {
                count += 1;
            }
        }
        return count;
    }
}

File: Homework/CSharp-Part-2/03. Methods/AppearanceCount.cs
using System;

class AppearanceCount
{
    static void Main()
    {
        int arraySize = int.Parse(Console.ReadLine());
        string input = Console.ReadLine();
        string[] elements = input.Split(new char[] { ' ' });

        int[] numbers = new int[arraySize];

        for (int i = 0; i < numbers.Length; i++)
        {
            numbers[i] = int.Parse(elements[i]);
        }

        int numberX = int.Parse(Console.ReadLine());

        Console.WriteLine(CountAppearances(numbers, numberX));
    }

    static int CountAppearances(int[] source, int target)
    {
        if (source == null)
        {
            throw new ArgumentNullException();
        }

        int count = 0;

        foreach (var item in source)
        {
            if (item == target)
            {
                ++count;
            }
        }

        return count;
    }
}